### PR TITLE
Enhance redis binary classifier to support additional versions

### PR DESCRIPTION
**Pull request by whalelines**
Allow existing matcher to match host identifiers longer than 12 characters. The binaries distributed by redis have the version before payload, so add a matcher for that. Add test fixtures covering these scenarios.

### CODE REVIEW

File: syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-7.0.14/redis-server
Comment (wagoodman) [inline syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-7.0.14/redis-server:1]: Can you reduce the size of these file to something that is as small as possible? Take for example the snippet from https://github.com/anchore/syft/blob/main/syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/perl-5.37.8/perl

Comment (whalelines) [reply syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-7.0.14/redis-server:1]: I certainly can. At present, they are within the range of the other redis "binaries".

```
$ ll syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-*/redis-server
-rwxr-xr-x 1 dd staff 7025 Oct 23 14:52 syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-2.8.23/redis-server
-rwxr-xr-x 1 dd staff 9179 Oct 23 14:52 syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-4.0.11/redis-server
-rwxr-xr-x 1 dd staff 7061 Oct 23 14:52 syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-5.0.0/redis-server
-rwxr-xr-x 1 dd staff 2574 Oct 23 14:52 syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-6.0.16/redis-server
-rwxr-xr-x 1 dd staff 1239 Oct 23 14:52 syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-7.0.0/redis-server
-rwxr-xr-x 1 dd staff 2919 Nov 15 16:35 syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-7.0.14/redis-server
-rwxr-xr-x 1 dd staff 3730 Nov 15 16:35 syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-7.2.3-amd64/redis-server
-rwxr-xr-x 1 dd staff 3494 Nov 15 16:35 syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-7.2.3-arm64/redis-server
```

Comment (whalelines) [reply syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-7.0.14/redis-server:1]: They are all under 1 kB now.

```
-rwxr-xr-x 1 dd staff  890 Nov 16 08:50 syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-7.0.14/redis-server
-rwxr-xr-x 1 dd staff  831 Nov 16 08:50 syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-7.2.3-amd64/redis-server
-rwxr-xr-x 1 dd staff  816 Nov 16 08:51 syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-7.2.3-arm64/redis-server
```

I have rebased to remove the larger versions from the history.

Comment (wagoodman) [reply syft/pkg/cataloger/binary/test-fixtures/classifiers/positive/redis-server-7.0.14/redis-server:1]: thanks for the extra effort -- we're trying to get better at capturing these fixtures

**Review by wagoodman — APPROVED** [review]

**Pull request merged**